### PR TITLE
[BLOCKED] fix overlap for ins

### PR DESCRIPTION
**Pull request by yunfeiguo**
- [ ] [RES-XXXX](https://binatechnologies.atlassian.net/browse/RES-XXXX)
- [ ] Blocking PRs: ADD HERE IF ANY
* People
  - [ ] Reviewers (>=1): ADD HERE
  - [ ] Merger (>=1): ADD HERE
  - [ ] FYI: ADD HERE IF ANY
- [ ] Release note (if needed)
- [ ] No code copied from outside Bina 
- [ ] Tests (Unit, Workflow, ITL & Gherkin) are passing **NOT PASSING**
- [ ] PR labels, milestones & assignees

# Summary

Briefly describe the changes and reasons for the changes.

### CODE REVIEW

File: src/main/java/com/bina/varsim/tools/simulation/VCF2diploid.java [hunk 1]
```diff
@@ -361,7 +361,7 @@ private boolean addVariant(final byte[] maskedSequence, final Sequence reference
         variants in the same haplotype should never overlap.
         Otherwise, they should be merged at first place.
          */
-        for (int p = position; p < position + referenceAlleleLength; p++) {
+        for (int p = position; p <= position + referenceAlleleLength; p++) {
```
Comment (marghoob) [inline src/main/java/com/bina/varsim/tools/simulation/VCF2diploid.java]: Wouldn't this flag deletion side by side with an insertion as overlapping? An insertion only overlaps if it is flanked by deletion on both sides.

Comment (yunfeiguo) [inline src/main/java/com/bina/varsim/tools/simulation/VCF2diploid.java]: you mean some this case?
```
1 3 A -
1 3 A AT
```?

Comment (marghoob) [inline src/main/java/com/bina/varsim/tools/simulation/VCF2diploid.java]: yeah, this case should be okay. it's pretty much a SNP.

**Comment by yunfeiguo** [comment]
see #150

**Pull request closed without merging**